docs: Update XDP drivers list

janvi01 opened the pull request:
Please ensure your pull request adheres to the following guidelines:

- [X] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [x] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)
- [x] Thanks for contributing!

<!-- Description of change -->

Fixes: #22867 

```release-note
Update XDP drivers support list in BPF docs
```

moved the Driver supporting XDP list to a dedicated RST fragment to include it from where it has to be listed.

janvi01 committed:
docs: Update list for generic XDP support in driver

Signed-off-by: janvi01 <janvibajo1@gmail.com>
Signed-off-by: Quentin Monnet <quentin@isovalent.com>

qmonnet reviewed (changes requested):
I agree with Anna, it's no longer necessary to move the table since we only have one list in the docs today. Sorry for misleading you about this earlier.

And I don't know why we would copy the list from the BCC docs, worst of all with no attribution.

What we need to do to address #22867 is to keep the current list and format, but update the list with the drivers who recently got support for XDP in the Linux kernel driver. This `grep` command you found should help you determine what drivers have support.

gentoo-root commented on Documentation/bpf/progtypes.rst:
```diff
@@ -338,49 +338,57 @@ corresponding network driver name of an interface can be determined as follows:
++-------------------+---------------+-------------+
+| Freescale         | dpaa2         | >= 5.0      |
++-------------------+---------------+-------------+
+| Generic XDP       |               | >= 4.12     |
```
The text above says:

> A list of drivers supporting native XDP can be found in the table below.

I think it's confusing to include generic XDP in the table of native XDP drivers (it's not native, and it's not a driver).

janvi01 replied:
ah yeah, better to not list it here. Thanks, updated.

gentoo-root reviewed (approved):
Looks good from my side. Please squash all commits before merging and update the release note in the pull request description.

qmonnet commented on Documentation/bpf/progtypes.rst:
```diff
@@ -338,49 +338,55 @@ corresponding network driver name of an interface can be determined as follows:
++-------------------+---------------+-------------+
+| Broadcom          | bnxt_en       | >= 4.11     |
++-------------------+---------------+-------------+
+| Cavium            | qed           | >= 4.10     |
```
Where does this one come from?

janvi01 replied:
QLogic (Cavium) qed* drivers | 4.10 | 496e05170958
-- | -- | --

gentoo-root replied:
> Where does this one come from?

QLogic is Cavium's acquisition. I'm not familiar with the structure of their driver (i.e. which subdirectories are based on which), but XDP parts are there in **qede** in 4.10:

https://elixir.bootlin.com/linux/v4.10/source/drivers/net/ethernet/qlogic/qede/qede_main.c#L2528

In the recent kernels it looks like XDP for PFs is supported by all subdrivers of **qede**, and XDP for VFs is only supported by **qed**:

https://elixir.bootlin.com/linux/v6.7/source/drivers/net/ethernet/qlogic/qed/qed_l2.c#L2254
https://elixir.bootlin.com/linux/v6.7/source/drivers/net/ethernet/qlogic/qed/qed_l2.c#L2273
https://elixir.bootlin.com/linux/v6.7/source/drivers/net/ethernet/qlogic/qede/qede_main.c#L856

gentoo-root replied:
> QLogic is Cavium's acquisition.

I wonder if we should rename Mellanox to NVIDIA, or stick with QLogic, or provide both names in both cases.

qmonnet replied:
> QLogic is Cavium's acquisition.

Ah right now I see, thanks Maxim.

But we already have an entry for QLogic, so adding Cavium creates a duplicate.

> I wonder if we should rename Mellanox to NVIDIA, or stick with QLogic, or provide both names in both cases.

Good question. I'd stick with the current names, that are also easier to map to kernel drivers, and change if the names change in the kernel tree?

gentoo-root replied:
> But we already have an entry for QLogic, so adding Cavium creates a duplicate.

Oh right, there is one down below. We don't need to add Cavium/qed then.

> I'd stick with the current names, that are also easier to map to kernel drivers, and change if the names change in the kernel tree?

Makes sense to me.

qmonnet commented on Documentation/bpf/progtypes.rst:
```diff
@@ -338,49 +338,55 @@ corresponding network driver name of an interface can be determined as follows:
++-------------------+---------------+-------------+
+| Intel             | ice           | >= 5.5      |
+|                   +---------------+-------------+
+|                   | igb           | >= 5.10     |
```
Looks good

qmonnet reviewed (changes requested):
There seems to be some drivers still missing, can you please complete the list?

qmonnet commented on Documentation/bpf/progtypes.rst:
```diff
@@ -338,49 +338,92 @@ corresponding network driver name of an interface can be determined as follows:
+|                        +-------------------------+-------------+
+|                        | Veth                    | >= 4.19     |
++------------------------+-------------------------+-------------+
+| Qlogic                 | qede                    | >= 4.10     |
```
```suggestion
| QLogic                 | qede                    | >= 4.10     |
```

qmonnet commented on Documentation/bpf/progtypes.rst:
```diff
@@ -338,49 +338,92 @@ corresponding network driver name of an interface can be determined as follows:
++------------------------+-------------------------+-------------+
+| Qlogic                 | qede                    | >= 4.10     |
++------------------------+-------------------------+-------------+
+| SFC                    | Efx                     | >= 5.5      |
```
Duplicate with Solarflare's sfc?

janvi01 replied:
yes, just checked.

qmonnet commented on Documentation/bpf/progtypes.rst:
```diff
@@ -338,49 +338,92 @@ corresponding network driver name of an interface can be determined as follows:
++------------------------+-------------------------+-------------+
+| Microchip              | lan966x                 | >= 6.1      |
++------------------------+-------------------------+-------------+
+| Microsoft              | hv_netvsc               | >= 5.6      |
```
```suggestion
| Microsoft              | hv_netvsc (Hyper-V)     | >= 5.6      |
```

qmonnet commented on Documentation/bpf/progtypes.rst:
```diff
@@ -338,49 +338,92 @@ corresponding network driver name of an interface can be determined as follows:
++------------------------+-------------------------+-------------+
+| VMware                 | vmxnet3                 | >= 6.6      |
++------------------------+-------------------------+-------------+
+| Xen                    | xennet                  | >= 5.9      |
```
Isn't that one the same as `xen-netfront` in the “Others” section?

janvi01 replied:
ah yes, both are same.

qmonnet reviewed (commented):
OK, looks like we're getting here, thanks!

Can you please squash all your commits into a single one, and force-push onto the branch?

dylandreimerink commented on Documentation/bpf/progtypes.rst:
```diff
@@ -338,49 +338,88 @@ corresponding network driver name of an interface can be determined as follows:
++========================+=========================+=============+
+| Amazon                 | ena                     | >= 5.6      |
++------------------------+-------------------------+-------------+
+| Aquantia               | atlantic                | >= 5.13     |
```
If I am not mistaken, support for atlantic was added in v5.19 with this PR https://github.com/torvalds/linux/commit/0d14657f40830243266f972766f1e4d00436e648

dylandreimerink commented on Documentation/bpf/progtypes.rst:
```diff
@@ -338,49 +338,88 @@ corresponding network driver name of an interface can be determined as follows:
++------------------------+-------------------------+-------------+
+| Freescale              | dpaa                    | >= 5.0      |
+|                        +-------------------------+-------------+
+|                        | dpaa2                   | >= 5.11     |
```
The versions are reversed dpaa2 got support in 5.0 [see](https://github.com/torvalds/linux/commit/7e273a8ebdd3b83f94eb8b49fc8ee61464f47cc2)
and dpaa got support in v5.11 [see](https://github.com/torvalds/linux/commit/86c0c196cbe48f844721783d9162e46bc35c0c5a)

dylandreimerink commented on Documentation/bpf/progtypes.rst:
```diff
@@ -338,49 +338,88 @@ corresponding network driver name of an interface can be determined as follows:
++------------------------+-------------------------+-------------+
+| Fungible               | fun                     | >= 5.18     |
++------------------------+-------------------------+-------------+
+| Google                 | gve                     | >= 5.19     |
```
If I am not mistaken support for gve was added in v6.4 [see](https://github.com/torvalds/linux/commit/75eaae158b1b7d8d5bde2bafc0bcf778423071d3)

dylandreimerink commented on Documentation/bpf/progtypes.rst:
```diff
@@ -338,49 +338,88 @@ corresponding network driver name of an interface can be determined as follows:
+|                        +-------------------------+-------------+
+|                        | mlx5                    | >= 4.9      |
++------------------------+-------------------------+-------------+
+| Microchip              | lan966x                 | >= 6.1      |
```
I believe for lan966x support landed in 6.2 [see](https://github.com/torvalds/linux/commit/6a2159be7604f5cdd7f574f4e0922f61e63c3f16)

dylandreimerink commented on Documentation/bpf/progtypes.rst:
```diff
@@ -338,49 +338,88 @@ corresponding network driver name of an interface can be determined as follows:
++------------------------+-------------------------+-------------+
+| Microsoft              | hv_netvsc (Hyper-V)     | >= 5.6      |
+|                        +-------------------------+-------------+
+|                        | mana                    | >= 6.0      |
```
If I am not mistaken mana was added in v5.17 [see](https://github.com/torvalds/linux/commit/ed5356b53f070dea5dff5a01b740561cb8222199)

dylandreimerink commented on Documentation/bpf/progtypes.rst:
```diff
@@ -338,49 +338,88 @@ corresponding network driver name of an interface can be determined as follows:
++------------------------+-------------------------+-------------+
+| Solarflare             | SFC Efx                 | >= 5.5      |
++------------------------+-------------------------+-------------+
+| STMicro                | stmmac                  | >= 5.3      |
```
Support for stmmac added in v5.13 [see](https://github.com/torvalds/linux/commit/5fabb01207a2d3439a6abe1d08640de9c942945f)

dylandreimerink reviewed (changes requested):
Sorry for the late review. I believe there are several incorrect version numbers. 

The version numbers in my comments are based on the commits where `ndo_bpf` hooks were added, which I believe to be accurate and the first branch which includes this commit.

Not sure which source was used for these existing numbers, perhaps my method is flawed, in which case I would like to be enlightened.

dylandreimerink reviewed (approved):
Thanks for the version fixes!

qmonnet committed:
docs: Update spelling exception list after XDP support list update

Adjust the list of spelling exceptions as a follow-up to the update of
the list of drivers with XDP support, in the previous commit.

Signed-off-by: Quentin Monnet <quentin@isovalent.com>
Signed-off-by: janvi01 <janvibajo1@gmail.com>

gentoo-root commented on Documentation/spelling_wordlist.txt:
```diff
@@ -93,6 +97,7 @@ Pfaff
 Polytechnique
 QCon
+QLogic
 Qlogic
```
Remove Qlogic if we renamed it to QLogic?

gentoo-root commented on Documentation/bpf/progtypes.rst:
```diff
@@ -338,49 +338,88 @@ corresponding network driver name of an interface can be determined as follows:
+|                        +-------------------------+-------------+
+|                        | xen-netfront            | >= 5.9      |
+|                        +-------------------------+-------------+
+|                        | Veth                    | >= 4.19     |
```
Why capitalize veth? I think it's normally all lower case letters.
